services/web/client/source/class/qxapp/component/filter/TextFilter.js

```diff
@@ -0,0 +1,56 @@
+/* ************************************************************************
+
+   qxapp - the simcore frontend
+
+   https://osparc.io
+
+   Copyright:
+     2019 IT'IS Foundation, https://itis.swiss
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+
+   Authors:
+     * Ignacio Pascual (ignapas)
+
+************************************************************************ */
+
+/**
+ * Label and TextField bundled together to make a simple text filter.
+ */
+qx.Class.define("qxapp.component.filter.TextFilter", {
+  extend: qxapp.component.filter.UIFilter,
+
+  /**
+   * Constructor for the TextFilter takes UIFilters mandatory params plus an optional translation id for its label.
+   *
+   * @extends qxapp.component.filter.UIFilter
+   *
+   * @param {string} [labelTr=Filter] Translation id used for the filter label.
+   */
+  construct: function(filterId, groupId, labelTr = "Filter") {
+    this.base(arguments, filterId, groupId);
+    this._setLayout(new qx.ui.layout.HBox());
+
+    this._add(new qx.ui.basic.Label(this.tr(labelTr) + ":").set({
```

Comment:
this has two problems ... first, the text "Filter" will not be picked up for translation as it does not appear as a string inside tr(...) and second as there is a string operation the text will not be switchable to a different language at runtime. I would not use a label in front of the filter but rather put a placeholder into the textfield.